src/AssetsPage/AssetsTable/index.jsx

```diff
@@ -226,24 +228,28 @@ export class AssetsTable extends React.Component {
   }
 
   render() {
-    return (!this.props.assetsList.length) ? (
-      <span>Loading....</span>
-    ) : (
-      <div>
-        {this.renderStatusAlert()}
-        <Table
-          columns={Object.keys(this.columns).map(columnKey => ({
-            ...this.columns[columnKey],
-            onSort: () => this.onSortClick(columnKey),
-          }))}
-          data={this.addSupplementalTableElements(this.props.assetsList)}
-          tableSortable
-          defaultSortedColumn="date_added"
-          defaultSortDirection="desc"
-        />
-        {this.renderModal()}
-      </div>
-    );
+    // TODO: Add UI for when there is nothing in the list and we have a status returned from the API
+    // TODO: http://fhtwd0.axshare.com/#g=1&p=files-and-uploads-empty
+    // Only show Loading when the list is empty AND the status from the APIs are empty
+    return (this.props.assetsList.length === 0 &&
+      Object.keys(this.props.assetsStatus).length === 0) ? (
+        <span>Loading....</span>
+      ) : (
+        <div>
+          {this.renderStatusAlert()}
+          <Table
+            columns={Object.keys(this.columns).map(columnKey => ({
+              ...this.columns[columnKey],
+              onSort: () => this.onSortClick(columnKey),
+            }))}
+            data={this.addSupplementalTableElements(this.props.assetsList)}
+            tableSortable
+            defaultSortedColumn="date_added"
+            defaultSortDirection="desc"
+          />
+          {this.renderModal()}
+        </div>
+      );
```

Comment:
I'm okay changing this to the `if/else` because we'll have other cases of what to render later that will also make this harder to read -- i.e. what to render when the list of assets that comes back is empty (because nothing was returned or because there was an error).  I can set that block that is being rendered to a variable and still keep the one return statement and return that variable at the end.

Reply:
@efischer19 Probably easier to read in the full version of this file than the PR view, but is this easier to read overall?

Reply:
Looks great!